console/v1/types_console_plugin.go

```diff
@@ -48,6 +48,118 @@ type ConsolePluginSpec struct {
+	// CSP header.
+	//
+	// Example:
+	//   ConsolePlugin A directives:
+	//     script-src: https://script1.com/, https://script2.com/
+	//     font-src: https://font1.com/
+	//
+	//   ConsolePlugin B directives:
+	//     script-src: https://script2.com/, https://script3.com/
+	//     font-src: https://font2.com/
+	//     img-src: https://img1.com/
+	//
+	//   Unified set of CSP directives, passed to the OpenShift web console server:
+	//     script-src: https://script1.com/, https://script2.com/, https://script3.com/
+	//     font-src: https://font1.com/, https://font2.com/
+	//     img-src: https://img1.com/
+	//
+	//   OpenShift web console server CSP response header:
+	//     Content-Security-Policy: default-src 'self'; base-uri 'self'; script-src 'self' https://script1.com/ https://script2.com/ https://script3.com/; font-src 'self' https://font1.com/ https://font2.com/; img-src 'self' https://img1.com/; style-src 'self'; frame-src 'none'; object-src 'none'
+	//
+	// +kubebuilder:validation:MaxItems=5
+	// +listType=map
+	// +listMapKey=directive
+	// +optional
+	ContentSecurityPolicy []ConsolePluginCSP `json:"contentSecurityPolicy"`
+}
+
+// DirectiveType is an enumeration of OpenShift web console supported CSP directives.
+// LoadType is an enumeration of i18n loading types.
+// +kubebuilder:validation:Enum:="DefaultSrc";"ScriptSrc";"ImgSrc";"StyleSrc";"FontSrc"
+type DirectiveType string
+
+const (
+	// DefaultSrc directive serves as a fallback for the other CSP fetch directives.
+	// For more information about the DefaultSrc directive, see:
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/default-src
+	DefaultSrc DirectiveType = "DefaultSrc"
+	// ScriptSrc directive specifies valid sources for JavaScript.
+	// For more information about the ScriptSrc directive, see:
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src
+	ScriptSrc DirectiveType = "ScriptSrc"
+	// ImgSrc directive specifies a valid sources of images and favicons.
+	// For more information about the ImgSrc directive, see:
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/img-src
+	ImgSrc DirectiveType = "ImgSrc"
+	// StyleSrc directive specifies valid sources for stylesheets.
+	// For more information about the StyleSrc directive, see:
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/style-src
+	StyleSrc DirectiveType = "StyleSrc"
+	// FontSrc directive specifies valid sources for fonts loaded using @font-face.
+	// For more information about the FontSrcdirective, see:
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/font-src
+	FontSrc DirectiveType = "FontSrc"
+)
+
+// CSPDirectiveValue is single value for a Content-Security-Policy directive.
+// Each directive value must have a maximum length of 1024 characters and must not contain
+// whitespace, commas, or semicolons.
```

Comment:
This must go into the godoc for the field that is using this type as well please